File: IntentPassing/app/src/main/java/com/example/intentpassing/MainActivity.java
package com.example.intentpassing;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnNext;
        btnNext = findViewById(R.id.btnNext);

        Intent iNext = new Intent(MainActivity.this, SecondActivity.class);

        iNext.putExtra("title","Home");
        iNext.putExtra("StudentName", "John");
        iNext.putExtra("RollNo", 11);


        btnNext.setOnClickListener(view -> {


            startActivity(iNext);
        });



    }
}

File: IntentPassingkt/app/src/main/java/com/example/intentpassingkt/MainActivity.kt
package com.example.intentpassingkt

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnMove.setOnClickListener {
//            val iNext = Intent(this, SecondActivity::class.java)
            startActivity(Intent(this, SecondActivity::class.java))
        }
    }
}